guides/resource-provisioning/servicebus-queues.md

```diff
@@ -0,0 +1,138 @@
+
+## Update local development environment
+
+Add the following environment variables to the `.env` file in the root of the microservice to use the developer Service Bus queues during local development:
+
+```
+MESSAGE_QUEUE_HOST=<INSERT_VALUE_FROM_AZURE_PORTAL>
+MESSAGE_QUEUE_PASSWORD=<INSERT_VALUE_FROM_AZURE_PORTAL>
+MESSAGE_QUEUE_SUFFIX=<DEVELOPER_INITIALS>
+MESSAGE_QUEUE_USER=RootManageSharedAccessKey
+```
+
+**Do not commit the `.env` file to the git repository so make sure it is added to the `.gitignore`**
+
+Values for `MESSAGE_QUEUE_HOST` and `MESSAGE_QUEUE_PASSWORD` will be found in the Azure Portal.
+
+The `<DEVELOPER_INITIALS>` should match those used in the name of the Service Bus queues created for local devlopment.
+
+## Update Docker Compose to use Service Bus environment variables
+
+Add the following environment variables to the `environment` section of the `docker-compose.yaml`:
+
+```
+MESSAGE_QUEUE_HOST: ${MESSAGE_QUEUE_HOST:-notset}
+MESSAGE_QUEUE_PASSWORD: ${MESSAGE_QUEUE_PASSWORD:-notset}
+MESSAGE_QUEUE_USER: ${MESSAGE_QUEUE_USER:-notset}
+```
+
+And for every queue you have created also add:
+
+```
+<IDENTIFIER>_QUEUE_ADDRESS: ffc-<workstream>-<identifier>-${MESSAGE_QUEUE_SUFFIX}
+```
+
+where `<workstream>` and `<identifier>` refer to those parts of the queue name described above.
+
+## Update microservice Helm chart
+
+Update the `ConfigMap` template of the Helm Chart (`helm/<REPO_NAME>/templates/config-map.yaml`) to include the environment variables for the message queue host and every queue you have created:
+
+```
+MESSAGE_QUEUE_HOST: {{ quote .Values.container.messageQueueHost }}
+<IDENTIFIER>_QUEUE_ADDRESS: {{ quote .Values.container.<identifier>QueueAddress }}
+```
+
+Then create default values for these in the `container` section of the Helm Chart values file (`helm/<REPO_NAME>/values.yaml`):
+
+```
+container:
+  messageQueueHost: dummy
+  <identifier>QueueAddress: <identifier>
+```
+
+## Add queue names to Azure App Configuration
+
+Azure App Configuration stores values required by the Jenkins CI pipelines. For each message queue create two key-value entries relating to the main branch and PR queue names via the Azure Portal:
+
+1. **Key**: `dev/container.<identifier>QueueAddress`; **Value**: `ffc-<workstream>-<identifier>-dev`
+2. **Key**: `dev/pr/container.<identifier>QueueAddress`; **Value**: `ffc-<workstream>-<identifier>-pr`
+
+where `<workstream>` and `<identifier>` refer to those parts of the queue name described above.
+
+**Do not** add a label to these App Configuration entries.
+
+## Add messaging code to the microservice
+
+Update your microservice code using the relevant Azure authentication and Service Bus SDKs for your language.
+
+Patterns for using a Service Bus in microservice code are outside of the scope of this guide. An example is shown below for a Node.js microservice, but please check current best practice with the FFC Platform Team.
```

Comment:
I think we should specify that teams must use the official Azure SDKs so that we can use Pod Identity and distributed tracing.

Reply:
Done.